### PR TITLE
allow user defined pull request number

### PR DESCRIPTION
**Pull request by xli12**
As discussed in Issue #365 , here is the pull request for the feature of allowing a user-defined pull request number.

If there is a user-defined pull request number, and the pull request number cannot be found from the context, it will use the user-defined PR number.

By the way: thanks for creating this GitHub Action! It helps us a lot.

### CODE REVIEW

**Comment by christophwitzko** [comment]
It would be amazing to get this merged 🙏

File: src/writeSummaryToPR.ts [hunk 1]
```diff
@@ -18,6 +19,11 @@ const writeSummaryToPR = async ({ summary, markerPostfix }: {
     pullRequestNumber = await getPullRequestNumberFromTriggeringWorkflow(octokit);
   }
 
+  if (!pullRequestNumber && userDefinedPrNumber) {
+    core.info(`User-defined pull-request number: ${userDefinedPrNumber}`);
+    pullRequestNumber = userDefinedPrNumber;
+  }
+
```
Comment (davelosert) [inline src/writeSummaryToPR.ts:26]: Imho, the `userDefinedPrNumber` should take precedence over the automatically acquired ones. 

At least, that's how I would expect it if I specifically put the PR Number in. Could you adjust the order her accordingly? 🙂

File: src/index.ts [hunk 2]
```diff
@@ -45,10 +46,18 @@ const run = async () => {
 	summary
 		.addRaw(`<em>Generated in workflow <a href=${getWorkflowSummaryURL()}>#${github.context.runNumber}</a></em>`)
 
+	let processedPrNumber: number | undefined = Number(prNumber);
+	core.info(`Received pull-request number: ${prNumber}`);
+
+	if (!Number.isSafeInteger(processedPrNumber) || processedPrNumber <= 0) {
+		processedPrNumber = undefined;
+	}
```
Comment (davelosert) [inline src/index.ts:54]: Could you put this entire check into the `src/inputs/option.ts`? 

It feels like input validation should happen there.
Apart from that, very nice check about the `safeInterger` here 👍🏻

File: README.md [hunk 3]
```diff
@@ -85,6 +85,7 @@ This action requires the `pull-request: write` permission to add a comment to yo
 | `file-coverage-mode`        | Defines how file-based coverage is reported. Possible values are `all`, `changes` or `none`.                                                                               | `changes`                                                                     |
 | `name`                      | Give the report a custom name. This is useful if you want multiple reports for different test suites within the same PR. Needs to be unique.                               | ''                                                                            |
 | `json-summary-compare-path` | The path to the json summary file to compare against. If given, will display a trend indicator and the difference in the summary. Respects the `working-directory` option. | undefined                                                                     |
+| `pr-number`                 | An optional, user-defined pull request number.                                                                                                                             | undefined                                                                     |
```
Comment (davelosert) [inline README.md:88]: According with my other comments to change the order of precedence, could you change this to the following suggestion:

```suggestion
| `pr-number`                 | The number of the PR to post a comment to (if any)                                                                                                                             | If in the context of a PR, the number of that PR.<br/> If in the context of a triggered workflow, the PR of the triggering workflow.                                                                    <br/>If no PR context is found, it defaults to `undefined` |
```

**Review by davelosert — CHANGES_REQUESTED** [review]
Thank you very much for this PR. Overall, it looks great and I only left a few minor comments to address.

Big thank you for also fixing some of the formatting. I really should add some ESLint and Prettier 😅

File: src/index.ts (hunk 2 above)
Comment (xli12) [reply src/index.ts]: Sure, the validation has now been moved to `option.ts`.

**Review by davelosert — APPROVED** [review]
Great work, thank you very much for this.
I'll merge and release this for now, as it has been sitting and waiting for long enough 🙂

**Pull request merged**